addons/mail/static/src/core/web/follower_list.js

```diff
@@ -27,6 +27,11 @@ export class FollowerList extends Component {
                 this.props.thread.loadMoreFollowers();
             }
         });
+        onMounted(() => {
+            if (this.props.load) {
+                this.props.load(this.props.thread, ["followers", "suggestedRecipients"]);
+            }
+        });
```

Comment:
I suppose the intend of `props.load` is to reset listing of followers on close?

Currently listing is on `thread.followers`, so this is saved in models which are totally persistent except if there are some explicit code to clear them.

What we want is save listing of followers per `FollowerList`, so the data should be stored/managed by the component. The data are structured (store data that becomes discuss JS models), so we need these models to be scoped by component, so that each component can managed listing and scrap on close as they want).

What you can do is have controller returning a `data` that can be simply feed in JS models with `this.store.insert()`, but then another entry to insert in the context of the component.

Some controllers like `/mail/inbox/messages` do that, with `data` being inserted in models globally but `messages` is contextually managed by the code calling this route. You can make the component insert the follower id data in a `Record.many("Follower")` that is fully managed by the component. For example, roughly:

```js
// follower_list_view_model.js
class FollowerListView extends Record {
    static id = "id";
    followers = Record.many("Follower");
}
FollowerListViewModel.register();

// follower_list.js
let NEXT_ID = 0;

class FollowerList extends Component {
    setup() {
        this.followerListView = this.store.FollowerListView.insert(NEXT_ID++);
        onWillDestroy(() => this.followerListView.delete());
    }
}
```

Fetching followers is managed by `FollowerList` component, `followers` entry in response contains data id of followers to add:

```js
const res = await rpc("fetch_followers");
this.store.insert(res.data);
this.followerListView.followers.add(...res.followers);
```

Reply:
Yes `props.load` is used to reset the follower list upon closing. Currently, follower data is stored in `thread.followers`, which persists in the models unless explicitly cleared. You suggested having a separate model  `FollowerListView` with `Record.many('followers')`, where I would insert followers into the store through the component.

However since the current implementation works without issues, could you help me understand the reasoning a bit more for managing the `FollowerListView` directly within the component instead of using the `Thread`? Is this intended to improve a specific aspect, such as performance or maintainability?

Reply:
If we display follower list more than once (e.g. if you have chatter and chat window, and we show follower list also in chatter), then the `Thread.followers` field is a global concern and one list affects the other, which could lead to strange behaviour and/or bugs (e.g. one has list opens, other clears, but then you cannot load more followers on 1st list unless you close it too, etc.).

The idea with `FollowerListView` is to make a JS model record per `FollowerList` component instances, so that each have their own field `followers` they manage themselves. This prevent 1 follower lists to bump with other follower lists.